hash_table8.hpp: Compilation error when instantiating hash_function()

```log
hash_table8.hpp:415:43: error: binding reference of type 'MapKeyHasher&' to 'const MapKeyHasher' discards qualifiers
[build]   415 |     HashT& hash_function() const { return _hasher; }
[build]       |                                           ^~~~~~~
```

Const function cannot return a reference to a member variable.

--- a/hash_table8.hpp
+++ b/hash_table8.hpp
@@ -416,7 +416,7 @@ public:
     size_type bucket_count() const { return _num_buckets; }
     float load_factor() const { return static_cast<float>(_num_filled) / ((float)_mask + 1.0f); }
 
-    HashT& hash_function() const { return _hasher; }
+    const HashT& hash_function() const { return _hasher; }
     EqT& key_eq() const { return _eq; }
 
     void max_load_factor(float mlf)

